### PR TITLE
executor: add SHUTDOWN command and privilege check

### PR DESCRIPTION
**Pull request by ekalinin**
### What problem does this PR solve? 

Add:
- `SHUTDOWN` command implementation
- `SHUTDOWN` privilege check
- Migrations for `mysql.user` table

Fix for #5046 (+ see pingcap/parser#580)

### What is changed and how it works?

TiDB now is able to handle `shutdown` command. Also added privilege check for this command.

When `shutdown` comes to server it sends to itself `SIGTERM` signal.
`SIGTERM` signal causes graceful shutdown of the server.

At the moment `ekalinin/parser` is used to be able to run tests.
Will revert `go.mod / go.sum` when pingcap/parser#580 will be merged.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: executor/simple.go [hunk 1]
```diff
@@ -1011,3 +1014,10 @@ func (e *SimpleExec) autoNewTxn() bool {
 	}
 	return false
 }
+
+func (e *SimpleExec) executeShutdown(s *ast.ShutdownStmt) error {
+	sessVars := e.ctx.GetSessionVars()
+	logutil.BgLogger().Info("execute shutdown statement", zap.Uint64("conn", sessVars.ConnectionID))
+	// sending signal to signal.SetupSignalHandler
+	return syscall.Kill(syscall.Getpid(), syscall.SIGTERM)
```
Comment (SunRunAway) [inline executor/simple.go]: Could we use https://godoc.org/os#Process.Kill to adapt to more platforms?

Comment (ekalinin) [reply executor/simple.go]: Done

Comment (ekalinin) [reply executor/simple.go]: Current server's log (we do not execute graceful shutdown):

```
[2019/10/21 10:24:18.395 +03:00] [INFO] [server.go:411] ["new connection"] [conn=1] [remoteAddr=127.0.0.1:49830]
[2019/10/21 10:24:21.370 +03:00] [INFO] [simple.go:1020] ["execute shutdown statement"] [conn=1]
[1]    20204 killed     ./bin/tidb-server
```

Current client's log:

```
mysql> shutdown;
ERROR 2006 (HY000): MySQL server has gone away
```

**Comment by SunRunAway** [comment]
Thank you @ekalinin
Please fix the CI build problem.

**Comment by ekalinin** [comment]
Guys,

This test is failed:

```
[2019-10-21T09:21:34.465Z]     c.Assert(len(mysql.Priv2UserCol), Equals, len(mysql.AllGlobalPrivs))

[2019-10-21T09:21:34.465Z] ... obtained int = 26

[2019-10-21T09:21:34.465Z] ... expected int = 25
```

The reason is:
- pingcap/parser#581

(`GrantPriv` is not in `AllGlobalPrivs` any more)

**Comment by ekalinin** [comment]
Fixed all failed tests related to pingcap/parser#581

**Comment by tsthght** [comment]
@SunRunAway I think this PR will result in inconsistent behavior with mysql. 
image

**Pull request merged**